shared/search/providers.test.ts

```diff
@@ -563,7 +629,12 @@ describe('search providers', () => {
 })
 
 function assertQuery(searchQuery: string, expectedTerms: string[]): void {
-    const actualTerms = searchQuery.split(' ').filter(part => !!part)
+    // Split terms in a way that preserved escaped spaces
+    const actualTerms = searchQuery
+        .replace(/\\ /g, '---')
+        .split(' ')
+        .filter(part => !!part)
+        .map(term => term.replace(/---/g, '\\ '))
```

Comment:
FYI `split()` supports a regex and JS supports lookbehind assertions so you can do this in a non-hacky way too :) not that it's important for the test
```suggestion
    const actualTerms = searchQuery.split(/(?<!\\) /).filter(part => !!part)
```
matches only spaces not preceded by a backslash

Reply:
Oh I tried to do a split with regex but a capturing group and I couldn't get it not to eat the \\.